### PR TITLE
ANDROID-11510 Compose Tabs

### PR DESCRIPTION
**Pull request by dpastor**
### :goal_net: What's the goal?
Implement compose tabs, following [specifications from design.](https://www.figma.com/file/tOb2OmRf6gnEK3fnSWPKOG/%F0%9F%94%B8-Tabs-Specs?node-id=309-3976&t=jTuZhrq5diuHn7sa-0)

### :construction: How do we do it?
* Implementation mostly based on material code, but changing its code to work as design specified.
* Single `Tabs` function provides tabs implementation.

### ☑️ Checks
- [x] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
- [x] 🖼️ Screenshots/Videos

https://user-images.githubusercontent.com/5360064/224015176-8cd6e671-73d6-4149-83d9-72cc608ec57a.mp4

- [x] Mistica App QR or download link

https://appcenter-filemanagement-distrib5ede6f06e.azureedge.net/4484ec7a-d2db-4304-a306-f975463c0938/MisticaCatalog-debug-1678362841.apk?sv=2019-02-02&sr=c&sig=FlW90VdTWoDzrklfQF6j2AP0ynhhV7zNW%2BvvH0AspHw%3D&se=2023-03-10T11%3A55%3A56Z&sp=r

- [ ] Reviewed by Mistica design team

### CODE REVIEW

File: library/src/main/java/com/telefonica/mistica/compose/tabs/Tabs.kt [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+            var tabMeasurables = subcompose(tabsSlot, composableTabs)
+            var tabPlaceables = tabMeasurables.map {
+                it.measure(constraints)
+            }
+
+            val tabsTotalWidth = tabPlaceables.fold(0) { sum, element ->
+                sum + element.width
+            }
+
+            if (tabsTotalWidth < maxWidth) {
```
Comment (dpastor) [inline library/src/main/java/com/telefonica/mistica/compose/tabs/Tabs.kt:167]: If tabs width measures do not fill total width, we need to try to split available space proportionally for them.

File: library/src/main/java/com/telefonica/mistica/compose/tabs/Tabs.kt [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+                tabsSlot = Slots.ProportionalSizeTabs
+                tabMeasurables = subcompose(tabsSlot, composableTabs)
+
+                val tabCount = tabMeasurables.size
+                val proportionalSizeTabWidth = (maxWidth / tabCount)
+
+                val anyTabLongerThanProportionalSizeTabWidth = tabPlaceables.any {
+                    it.width > proportionalSizeTabWidth
+                }
+                tabPlaceables = if (anyTabLongerThanProportionalSizeTabWidth) {
```
Comment (dpastor) [inline library/src/main/java/com/telefonica/mistica/compose/tabs/Tabs.kt:177]: If any tab is longer than proportional size calculated, we split remaining space available between all tabs

File: library/src/main/java/com/telefonica/mistica/compose/tabs/Tabs.kt [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+                    val resizableTabsCount = tabPlaceables.count { element ->
```
Comment (dpastor) [inline library/src/main/java/com/telefonica/mistica/compose/tabs/Tabs.kt:178]: We do not give any extra space to tabs already having maximum tab width.

**Review by jeslat — APPROVED** [review]
good job! 👏

**Review by aweell — APPROVED** [review]
Great work! LGTM ✨

**Review by yceballost — APPROVED** [review]
😍

**Review by gmerinojimenez — APPROVED** [review]
Good job!!

**Pull request merged**